fetch/client/src/main/scala/endpoints4s/fetch/Endpoints.scala

```diff
@@ -281,68 +283,75 @@ trait EndpointsWithCustomErrors
       request: Request[A],
       response: Response[B],
       a: A
-  )(
-      onload: Either[Throwable, B] => Unit,
-      onerror: Throwable => Unit
-  ): Unit = {
+  ): (FutureLike[B], js.Function0[Unit]) = {
```

Comment:
I am sorry if I am a bit nitpicky here, but I would rather not introduce the `FutureLike` abstraction. I prefer staying with callbacks here (and every concrete interpreter turns these callbacks into their `Future` thing).

So, the signature of `performFetch` would be the following:

~~~ scala
protected final def performFetch[A, B](
  request: Request[A],
  response: Response[B],
  a: A
)(
  onload: Either[Throwable, B] => Unit,
  onerror: Throwable => Unit
): js.Function0[Unit]
~~~

Reply:
The reason why I went for the callback refactoring was `Thenable` interpreters - I could not come up with a pretty solution for them. I came up with this:
```scala
var outerResolve: js.UndefOr[js.Function1[B | js.Thenable[B], _]] = js.undefined
var outerError: js.UndefOr[js.Function1[scala.Any, _]] = js.undefined

val promise = new js.Promise[B]((resolve, error) => {
  outerResolve = resolve
  outerError = error
})

val jsAbort = performFetch(this.request, this.response, a)(
  _.fold(
    exn => outerError.asInstanceOf[js.Function1[scala.Any, _]].apply(exn.getMessage),
    b => outerResolve.asInstanceOf[js.Function1[B | js.Thenable[B], _]].apply(b)
  ): Unit,
  ex => outerError.asInstanceOf[js.Function1[scala.Any, _]].apply(ex): Unit
)

new Result(promise) { val abort = jsAbort }
```
I can implement it if it is fine, otherwise I would need some suggestions.

Reply:
Ah, indeed that’s because of how JS promises work. I am not sure we can do it without using an uninitialized variable…

I have pushed 88f5f37 with a design that I like for the xhr interpreter. What do you think? Do you think that would be appropriate for the fetch interpreter as well?

Reply:
Either the result initialization or the abort callback needs to be moved to a mutable variable(s), I chose the former and you chose the latter 😄 So we can just go your route with Fetch interpreter too.